.github/workflows/pypi_publish.yml

```diff
@@ -48,8 +56,104 @@ jobs:
           name: wheels-sdist
           path: ./wheelhouse/*.tar.gz
 
+  build-wheels:
+    name: 'Wheels: ${{ matrix.os }} ${{ matrix.arch }} ${{ matrix.python-version }}'
+    runs-on: ${{ matrix.os }}
+    continue-on-error: true
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: ['3.12']
+        include:
+          - os: ubuntu-latest
+            arch: x86_64
+          # - os: windows-latest
+          #   arch: AMD64
+          - os: macos-latest
+            arch: arm64
+
+    steps:
+    - name: Check out the repo
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0  # get the non-merge commit for PRs
+
+    # Install openmpi on macos
+    - name: Install openmpi and hdf5 on macos
+      if: matrix.os == 'macos-latest'
+      run: |
+        brew install open-mpi
+        echo "PKG_CONFIG_PATH=$(brew --prefix)/lib/pkgconfig:$(brew --prefix)/opt/open-mpi/lib/pkgconfig:$PKG_CONFIG_PATH" >> $GITHUB_ENV
+        echo "CPATH=$(brew --prefix)/include:$CPATH" >> $GITHUB_ENV
+        echo "LIBRARY_PATH=$(brew --prefix)/lib:$LIBRARY_PATH" >> $GITHUB_ENV
+
+    # Set up python on ubuntu (build runs in containers, so can't use conda)
+    # Can't use conda on macos, the binaries it installs are invalid according
+    # to meson: "executables created by cpp compiler are not runnable"
+    - name: 🐍 Set up Python
+      if: matrix.os == 'ubuntu-latest' || matrix.os == 'macos-latest'
+      uses: actions/setup-python@v5
+
+    - name: Set python include path on macos
+      if: matrix.os == 'macos-latest'
+      run: |
+          echo "CPATH=$(python -c 'import sysconfig; print(sysconfig.get_path("include"))'):$CPATH" >> $GITHUB_ENV
+
+    # Use environment-windows.yml because windows only has msmpi, not openmpi
+    - name: 🐍 Set up Python
+      if: matrix.os == 'windows-latest'
+      uses: conda-incubator/setup-miniconda@v3
+      with:
+        miniconda-version: "latest"
+        activate-environment: build-vhdf
+        environment-file: environment-windows.yml
+        conda-remove-defaults: true
```

Comment:
What happens if you just use `setup-python` in Windows?
Also the whole windows path is currently commented out at the top of the file